components/inline/src/vwc-inline.scss

```diff
@@ -13,49 +14,82 @@ $items-min-inline-sizes: (
 );
 
 @each $key, $value in $items-min-inline-sizes {
-	:host([size='#{$key}'i]) {
-		#{$items-min-inline-size-v-name}: $value;
+	:host([column-basis="#{$key}"i]) {
+		.layout {
+			#{$items-min-inline-size-v-name}: $value;
+		}
 	}
 }
 /* #endregion */
 
 /* #region SPACING */
 $items-spacings: (
-	// ! hardcoded values should be defined and auto-generated to, and by, design tokens
-	'sm': 24px,
-	'md': 32px
+	
+	'xs': 16px,
+	'md': 24px,
+	'xl': 32px
 );
 
 @each $key, $value in $items-spacings {
-	:host([spacing='#{$key}'i]) {
-		#{$items-spacing-v-name}: $value;
+	:host([column-spacing="#{$key}"i]) {
+		.layout {
+			#{$items-spacing-v-name}: $value;
+		}
+	}
+}
+/* #endregion */
+
+/* #region GUTTERS */
+$items-gutters: (
+	
+	'xs': 16px,
+	'md': 24px,
+	'xl': 32px
+);
+
+@each $key, $value in $items-gutters {
+	:host([inline-gutters="#{$key}"i]) {
+		.layout {
+			#{$inline-gutters-v-name}: $value;
+		}
 	}
 }
 /* #endregion */
 
 :host {
-	display: grid;
-	width: 100%;
-	gap: var(#{$items-spacing-v-name}, #{map.get($items-spacings, md)});
-	grid-auto-rows: min-content;
-	grid-template-columns: repeat(
-		var(#{$inline-template}),
-		minmax(
-			var(
-				#{$items-min-inline-size-v-name},
-				#{map.get($items-min-inline-sizes, lg)}
-			),
-			1fr
-		)
-	);
-	inline-size: 100%;
+	.layout {
+		display: grid;
+		margin: var(#{$inline-gutters-v-name}, #{map.get($items-gutters, md)});
+		gap: var(#{$items-spacing-v-name}, #{map.get($items-spacings, md)});
+		grid-auto-rows: min-content;
+		grid-template-columns: repeat(
+			var(#{$inline-template}),
+			minmax(var(#{$items-min-inline-size-v-name}, #{map.get($items-min-inline-sizes, lg)}), 1fr)
+		);
+	}
+}
+
+:host([inline-gutters='']) {
+	.layout {
+		margin: 0;
+	}
+}
+
+:host([column-basis="block"i]) {
+	.layout {
+		grid-template-columns: 1fr;
+	}
 }
 
 :host(:not(template)),
 :host([template="fit"i]) {
-	#{$inline-template}: auto-fit;
+	.layout {
```

Comment:
there's a lot of repetitive `.layout` selector.
in sass u can parent select
e.g.
```css
.layout {
  :host(:not(template)) &,
  :host([template="fit"i]) & {
    #{$inline-template}: auto-fit;
  }
  // ... and all other parent selectors
}
```